### PR TITLE
http3: add request to response

### PR DESCRIPTION
**Pull request by Davincible**
fix: #3605

### CODE REVIEW

File: http3/client_test.go [hunk 1]
```diff
@@ -741,6 +741,9 @@ var _ = Describe("Client", func() {
 			Expect(rsp.Proto).To(Equal("HTTP/3.0"))
 			Expect(rsp.ProtoMajor).To(Equal(3))
 			Expect(rsp.StatusCode).To(Equal(418))
+			Expect(rsp.Request).ToNot(Equal(nil))
```
Comment (marten-seemann) [inline http3/client_test.go:744]:
```suggestion
			Expect(rsp.Request).ToNot(BeNil())
```

File: http3/client.go [hunk 2]
```diff
@@ -251,6 +251,10 @@ func (c *client) RoundTripOpt(req *http.Request, opt RoundTripOpt) (*http.Respon
 		return nil, c.handshakeErr
 	}
 
+	req.Proto = "HTTP/3.0"
+	req.ProtoMajor = 3
+	req.ProtoMinor = 0
```
Comment (marten-seemann) [inline http3/client.go:256]: Why do we need to set these? The documentation for `http.Request` says:
> For client requests, these fields are ignored.

Comment (Davincible) [reply http3/client.go:256]: > If we return it, it should be set correctly though

Otherwise, the request added to the response will say it's http1

Comment (marten-seemann) [reply http3/client.go:256]: I was just wondering what happens if you use the same `http.Request` struct multiple times, potentially on HTTP/3 and HTTP/2. Are we creating a race condition here by modifying the struct?

File: http3/client.go [hunk 3]
```diff
@@ -11,12 +11,13 @@ import (
...
 	"time"
 
+	"github.com/marten-seemann/qpack"
+
 	"github.com/lucas-clemente/quic-go"
 	"github.com/lucas-clemente/quic-go/internal/protocol"
 	"github.com/lucas-clemente/quic-go/internal/qtls"
 	"github.com/lucas-clemente/quic-go/internal/utils"
 	"github.com/lucas-clemente/quic-go/quicvarint"
-	"github.com/marten-seemann/qpack"
```
Comment (marten-seemann) [inline http3/client.go:19]: Can you revert this one?

Comment (Davincible) [reply http3/client.go]: fixed

**Review by marten-seemann — APPROVED** [review]
Thank you!

**Pull request merged**